src/map/attack.cpp

```diff
@@ -566,6 +566,67 @@ bool CAttack::CheckCover()
  ************************************************************************/
 void CAttack::ProcessDamage()
 {
+    if (settings::get<bool>("map.ENABLE_AUTO_ATTACK_LUA"))
+    {
+        // Sneak attack.
+        if (m_attacker->GetMJob() == JOB_THF && m_isFirstSwing && m_attacker->StatusEffectContainer->HasStatusEffect(EFFECT_SNEAK_ATTACK) &&
+            (behind(m_attacker->loc.p, m_victim->loc.p, 64) || m_attacker->StatusEffectContainer->HasStatusEffect(EFFECT_HIDE) ||
+             m_victim->StatusEffectContainer->HasStatusEffect(EFFECT_DOUBT)))
+        {
+            m_isSA = true;
+        }
+
+        // Trick attack.
+        if (m_attacker->GetMJob() == JOB_THF && m_isFirstSwing && m_attackRound->GetTAEntity() != nullptr)
+        {
+            m_isTA = true;
+        }
+
+        auto calculateAttackDamage = lua["xi"]["combat"]["physical"]["calculateAttackDamage"];
+        auto result                = calculateAttackDamage(m_attacker, m_victim, GetWeaponSlot(), m_attackType, m_attackRound->IsH2H(), m_isFirstSwing, m_isSA, m_isTA, m_damageRatio, m_attacker->GetMainWeaponDmg());
```

Comment:
This is currently hardcoded to only use  m_attacker->GetMainWeaponDmg(), but this could be fixed here or in lua